### PR TITLE
[WIP] - Storable

### PR DESCRIPTION
**Pull request by LoganWright**
- Initial storable protocol
- Reference type move
- Removing all instances of `"id"` related to models. Avoid using anything besides `idKey` when relating to models

### CODE REVIEW

File: Sources/Fluent/Entity/Entity.swift [hunk 1]
```diff
@@ -1,9 +1,51 @@
...
+    var storage: Storage { get }
+}
+
+extension Storable {
+    /// Whether or not entity was retrieved from database.
+    ///
+    /// This value shouldn't be interacted w/ external users
+    /// w/o explicit knowledge.
+    ///
+    /// General implementation should just be `let storage = Storage()`
```
Comment (tanner0101) [inline Sources/Fluent/Entity/Entity.swift]: This last line of the comment seems like it should be on the protocol.

File: Sources/Fluent/Entity/Entity.swift [hunk 2]
```diff
@@ -1,9 +1,51 @@
...
+            storage.exists = newValue
+        }
+    }
+
+    public var id: Node? {
+        get {
+            return storage.id
+        }
+        set {
+            // TODO: Should we log errs if storage.id already has value?
```
Comment (tanner0101) [inline Sources/Fluent/Entity/Entity.swift]: Overwriting ids is normal. Sometimes you need to pull a user out, change its id, and re-save.

File: Sources/Fluent/Entity/Entity.swift [hunk 3]
```diff
@@ -68,8 +110,8 @@ extension Entity {
 extension Entity {
     /// Persists the entity into the
     /// data store and sets the `id` property.
-    public mutating func save() throws {
-        try Self.query().save(&self)
+    public func save() throws {
+        try Self.query().save(self)
```
Comment (tanner0101) [inline Sources/Fluent/Entity/Entity.swift]: 😍

File: Sources/Fluent/Entity/Entity.swift [hunk 4]
```diff
@@ -1,9 +1,51 @@
...
+        set {
+            // TODO: Should we log errs if storage.id already has value? 
+            // Users shouldn't overwrite id
+            storage.id = newValue
+        }
+    }
+}
+
+extension Entity {
+    public var idKey: String {
```
Comment (tanner0101) [inline Sources/Fluent/Entity/Entity.swift]: This should go with the other idKey conveniences in the Entity.swift file.

File: Sources/Fluent/Entity/Entity.swift (hunk 2 above)
Comment (LoganWright) [inline Sources/Fluent/Entity/Entity.swift]: Should that delete the old user? Either way, sounds like onus is on end user, I'll remove the comment 👍

Comment (tanner0101) [inline Sources/Fluent/Entity/Entity.swift]: Hrm.... I guess that's true. It's normally like `UPDATE users SET id = 42 WHERE id = 3` but that wouldn't quite work here. If this becomes a problem we could make it work with some internal `originalId`.

**Pull request merged**